lib/object.gi

```diff
@@ -763,27 +763,44 @@ InstallGlobalFunction( MU_Finalize, function (  )
   fi;
 end );
 
-InstallMethod( MemoryUsage, "fallback method for objs without subobjs",
+InstallMethod( MemoryUsage, "generic fallback method",
   [ IsObject ],
   function( o )
-    local mem;
-    mem := SHALLOW_SIZE(o);
-    if mem = 0 then 
+    local mem,known,i,s;
+
+    if SHALLOW_SIZE(o) = 0 then
         return MU_MemPointer;
+    fi;
+
+    if MU_AddToCache( o ) then
+        return 0;    # already counted
+    fi;
+
+    MEMUSAGECACHE.depth := MEMUSAGECACHE.depth + 1;
+
+    # Count the bag, the header, and the master pointer
+    mem := SHALLOW_SIZE(o) + MU_MemBagHeader + MU_MemPointer;
+    if IS_POSOBJ(o) then
+        # Again the bag, its header, and the master pointer
+        for i in [1..LEN_POSOBJ(o)] do
+            if IsBound(o![i]) then
+                if SHALLOW_SIZE(o![i]) > 0 then    # a subobject!
+                    mem := mem + MemoryUsage(o![i]);
+                fi;
+            fi;
+        od;
+    elif IS_COMOBJ(o) then
+        # Again the bag, its header, and the master pointer
+        for i in NamesOfComponents(o) do
+            s := o!.(i);
+            if SHALLOW_SIZE(s) > 0 then    # a subobject!
+                mem := mem + MemoryUsage(s);
+            fi;
+        od;
     else
-        # a proper object, thus we have to add it to the database
-        # to not count it again!
-        if not(MU_AddToCache(o)) then
-            mem := mem + MU_MemBagHeader + MU_MemPointer;
-            # This is for the bag, the header, and the master pointer
-        else 
-            mem := 0;   # already counted
-        fi;
-        if MEMUSAGECACHE.depth = 0 then   
-            # we were the first to be called, thus we have to do the cleanup
-            ClearObjectMarker( MEMUSAGECACHE );
-        fi;
+        # objs without subobjs
```

Comment:
@frankluebeck You mean this clause?

I don't mind deleting it, but I had a reason why I left it in: Namely as a reminder that there can be other object types... Though the comment is indeed misleading! For such objects can actually contain subobjects, we just don't have a way to access them! For example, in SingularInterface and NormalizInterface we do this. Perhaps an `InforWarning` would be appropriate? But then we'd have to take care to only show it for external objects

So, should I remove this? Or instead change it to something like this:
```
elif TNUM_OBJ_INT(o) >= FIRST_EXTERNAL_TNUM then
  Info( InfoWarning, 1, "MemoryUsage encountered an external object, results may be too low" );
```
Note that I do not suggest to use `TryNextMethod` because this is already the fallback method.